### PR TITLE
feat: Token safety labels/speedbumps

### PR DESCRIPTION
**Pull request by cartcrom**
Created token safety components & modal, linked them into existing import speedbumps on /swap + token selector & added them to Token Details page.

Once a token warning is acknowledged, it is imported and the same warning should not appear again (unless you unimport the token)

To view swap speedbump:
* Navigate to a URL like /#/swap?outputCurrency=0x0cEC1A9154Ff802e7934Fc916Ed7Ca50bDE6844e&inputCurrency=0x1796ae0b0fa4862485106a0de9b654eFE301D0b2&chain=mainnet

To view token selector speedbump: 
* Make sure Uni default list is enabled, and Uni extended list is not
* import a medium warning token like 0x0cEC1A9154Ff802e7934Fc916Ed7Ca50bDE6844e
* or strong warning: 0x30ceCB5461A449A90081F5a5F55db4e048397BAB
* or an unsupported warning: 0x4922a015c4407F87432B179bb209e125432E4a2A

To view token details speedbump & warning messages:
* Navigate to explore and select a non-default token like PoolTogether

### CODE REVIEW

File: src/components/TokenSafety/index.tsx [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  display: block;
+  text-decoration: none;
+  font-size: 14px;
+  color: ${({ theme }) => theme.accentAction};
+  overflow: hidden;
+  text-overflow: ellipsis;
+`
+
+function EtherscanURL({ tokenAddress }: { tokenAddress: string | null | undefined }) {
+  const learnMoreURL = 'https://etherscan.io/token/' + tokenAddress
```
Comment (cmcewen) [inline src/components/TokenSafety/index.tsx:142]: does this need to be different per network?

File: src/constants/TokenSafetyLookupTable.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  constructor() {
+    if (!this.uniDefaultTokens) {
+      this.uniDefaultTokens = []
+    }
+    if (!this.uniExtendedTokens) {
+      this.uniExtendedTokens = []
+    }
+
+    this.brokenTokens.forEach((token) => {
```
Comment (cmcewen) [inline src/constants/TokenSafetyLookupTable.ts:30]: i worry about a future where these token lists grow large and this is a lot of startup initialization. the best idea i have around that is do make `checkToken` async and lazily initialize. wdyt?

File: src/constants/tokenWarnings.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        heading: <Trans>{"These tokens aren't verified"}</Trans>,
+        description: <Trans>{'Please do your own research before trading'}.</Trans>,
+      }
+    }
+    return {
+      heading: <Trans>{"This token isn't verified"}</Trans>,
+      description: <Trans>{'Please do your own research before trading.'}</Trans>,
+    }
+  } else {
+    if (plural) {
```
Comment (cmcewen) [inline src/constants/tokenWarnings.tsx:27]: fyi lingui has some tools for this https://lingui.js.org/guides/plurals.html

**Review by cmcewen — COMMENTED** [review]
overall this is looking pretty good! nice work

**Comment by JFrankfurt** [comment]
image
It looks in the designs these etherscan links are supposed to be truncated instead of wrapping
image

**Comment by JFrankfurt** [comment]
double check about a help center link for "learn more" 
image

**Comment by JFrankfurt** [comment]
Talk to @infredible about:

- using the `CopyHelper` component to replace the copied text
- changing the `CopyHelper` success icon to a square check
- making the link text clickable

File: src/constants/TokenSafetyLookupTable.ts (hunk 2 above)
Comment (cartcrom) [reply src/constants/TokenSafetyLookupTable.ts:30]: Yeah forcing it to run through an extra couple iterations makes it take like 3 seconds. Maybe there is a way to have redux initialize the table as its initializing itself? Or maybe redux only unpacks the default list and doesn't unpack extended... I'll look into it, but curious to get your thoughts @cmcewen

File: src/components/TokenSafety/TokenSafetyMessage.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      <TitleRow>
+        {warning.canProceed ? <AlertTriangle size={'16px'} /> : <AlertOctagon size={'16px'} />}
+        <Title marginLeft="7px">{warning.message}</Title>
+      </TitleRow>
+
+      <DetailsRow>
+        {heading && [heading, '. ']}
+        {description}
+        {tokenAddress && (
+          <ExternalLink href={'https://etherscan.io/token/' + tokenAddress}>
```
Comment (cmcewen) [inline src/components/TokenSafety/TokenSafetyMessage.tsx:56]: `getExplorerLink` here too?

**Review by cmcewen — APPROVED** [review]
one more nit - after you get sign off from fred, go ahead and merge

**Pull request merged**